File: 20200515-java-password-validator/src/main/java/com/katas/PasswordValidator.java
package com.katas;

public class PasswordValidator {
    public boolean isValid(String password) {
        return password.matches(".*[A-Z].*") &&
                password.matches(".*[a-z].*") &&
                password.matches(".*[\\d].*") &&
                password.contains("_") &&
                password.length() >= 8;
    }
}

File: 20240731-ts-password-validator/app/password.validator.ts
export class PasswordValidator {
  isValid(password: string) {
    return this.validateLength(password)
      && this.validateContainsCapital(password)
      && this.validateContainsLowerCase(password)
      && this.validateContainsUnderscore(password);
  }

  private validateContainsUnderscore(password: string) {
    return password.includes("_");
  }

  private validateContainsCapital(password: string) {
    return /[A-Z]/.test(password);
  }

  private validateContainsLowerCase(password: string) {
    return /[a-z]/.test(password);
  }

  private validateLength(password: string) {
    return password.length >= 8;
  }
}
